Improve logging, remove stale codepaths, fix upsync issues

chriadam opened the pull request.

VDVsx commented:
Btw, I've been thinking about switch the buteo own logging framework to new Qt logging(http://doc.qt.io/qt-5/qmessagelogger.html), current logging is not that flexible IMO, would be also nice to have ability to send it to a file for easier debugging.

antseppa commented:
LGTM. Tried on a device as well.

chriadam commented:
I've updated this - added a new commit on top of the old two.
Note that I also modified the commit messages of the old two commits, but their content remains unchanged, so only the new commit is new code.
